### ISSUE TITLE
possible bug in AntecedentEstimator.determineMF()

### ISSUE THREAD

**Issue by tdhooghe** [CaroFuchs/pyFUME#3]
Dear Caro,

When I am trying to run the following code 

```
def train_final_model(mf_shape, variable_names, nr_clus):
    all_variable_names = variable_names + ['Price']
    # load data based on selected features
    dl = DataLoader(dataframe=house_data[all_variable_names], normalize='minmax', verbose=False)

    # use all available data from now on to train the model
    cl = Clusterer(nr_clus=nr_clus, x_train=dl.dataX, y_train=dl.dataY, verbose=False)
    _, partition_matrix, _ = cl.cluster(method="fcm")  # 'fst-pso' or 'fcm' or 'gk'

    ae = AntecedentEstimator(dl.dataX, partition_matrix)
    antecedent_parameters = ae.determineMF(mf_shape=mf_shape)

    fsc = FireStrengthCalculator(antecedent_parameters, nr_clus=nr_clus, variable_names=variable_names)
    firing_strengths = fsc.calculate_fire_strength(dl.dataX)

    ce = ConsequentEstimator(dl.dataX, dl.dataY, firing_strengths=firing_strengths)
    consequent_parameters = ce.suglms()

    model_builder = SugenoFISBuilder(
        antecedent_parameters,
        consequent_parameters,
        variable_names,
        model_order="first",
        extreme_values=None,
        save_simpful_code=True,
    )
    return model_builder.get_model()


final_model = train_final_model(mf_shape='sigmf', variable_names=variable_names, nr_clus=9)
```
I get the following traceback
```
Traceback (most recent call last):
  File "/home/thomas/Documents/1BM120_computational_intelligence/py39env/lib/python3.9/site-packages/IPython/core/interactiveshell.py", line 3437, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-63-8da454315515>", line 30, in <module>
    final_model = train_final_model(mf_shape='sigmf', variable_names=variable_names, nr_clus=9)
  File "<ipython-input-63-8da454315515>", line 11, in train_final_model
    antecedent_parameters = ae.determineMF(mf_shape=mf_shape)
  File "/home/thomas/Documents/1BM120_computational_intelligence/py39env/lib/python3.9/site-packages/pyfume/EstimateAntecendentSet.py", line 57, in determineMF
    prm = self._fitMF(x=xx, mf=mf, mf_shape=mf_shape)
  File "/home/thomas/Documents/1BM120_computational_intelligence/py39env/lib/python3.9/site-packages/pyfume/EstimateAntecendentSet.py", line 284, in _fitMF
    param, _ = curve_fit(self._sigmf, x, mf, p0 = [c, s])
  File "/home/thomas/Documents/1BM120_computational_intelligence/py39env/lib/python3.9/site-packages/scipy/optimize/minpack.py", line 789, in curve_fit
    raise RuntimeError("Optimal parameters not found: " + errmsg)
RuntimeError: Optimal parameters not found: Number of calls to function has reached maxfev = 600.
```
This does not happen if "mf_shape='gauss'". When changing the mf_shape to 'gauss2', I get the following traceback: 
```
Traceback (most recent call last):
  File "/home/thomas/Documents/1BM120_computational_intelligence/py39env/lib/python3.9/site-packages/IPython/core/interactiveshell.py", line 3437, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-64-86c2a6b4eb2d>", line 30, in <module>
    final_model = train_final_model(mf_shape='gauss2', variable_names=variable_names, nr_clus=9)
  File "<ipython-input-64-86c2a6b4eb2d>", line 11, in train_final_model
    antecedent_parameters = ae.determineMF(mf_shape=mf_shape)
  File "/home/thomas/Documents/1BM120_computational_intelligence/py39env/lib/python3.9/site-packages/pyfume/EstimateAntecendentSet.py", line 57, in determineMF
    prm = self._fitMF(x=xx, mf=mf, mf_shape=mf_shape)
  File "/home/thomas/Documents/1BM120_computational_intelligence/py39env/lib/python3.9/site-packages/pyfume/EstimateAntecendentSet.py", line 273, in _fitMF
    param, _ = curve_fit(self._gauss2mf, x, mf, p0 = [mu1, sig1, mu2, sig2], bounds=((-np.inf, 0), (np.inf, np.inf)), maxfev=1000)
  File "/home/thomas/Documents/1BM120_computational_intelligence/py39env/lib/python3.9/site-packages/scipy/optimize/minpack.py", line 795, in curve_fit
    res = least_squares(func, p0, jac=jac, bounds=bounds, method=method,
  File "/home/thomas/Documents/1BM120_computational_intelligence/py39env/lib/python3.9/site-packages/scipy/optimize/_lsq/least_squares.py", line 798, in least_squares
    raise ValueError("Inconsistent shapes between bounds and `x0`.")
ValueError: Inconsistent shapes between bounds and `x0`.
```

For me it looks like unexpected behavior, but I might be missing something? 

Kind regards,
Thomas

**Comment by CaroFuchs** [comment]
Dear Thomas,

The RuntimeError you encountered when fitting the sigmoidal curves occurs when a sigmoidal curve simply does not fit the data and pyFUME cannot find suitable parameters. I've changed the error message that is printed to the console now to state the problem more clearly, and to suggest the user to try a different shape for the membership functions. 

The boundaries for the parameters of the double gaussian curve were indeed too strict. I changed this in the newest pyFUME release, so if you pip upgrade now your code should work.

Kind regards,

Caro
